pkg/server/server.go

```diff
@@ -673,6 +673,27 @@ func (s *Server) indexHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// This Content Security Policy (CSP) applies to Console web application resources.
+	// Console CSP is deployed in report-only mode via "Content-Security-Policy-Report-Only" header.
+	// See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP for details on CSP specification.
+	cspSources := "'self'"
+	if s.K8sMode == "off-cluster" {
+		// Console local development involves a webpack server running on port 8080
+		cspSources = cspSources + " http://localhost:8080 ws://localhost:8080"
+	}
```

Comment:
We should determine if there is a more precise way to determine if we're using a webpack dev server in development mode. There are other use cases I know of where people run an off-cluster console.

Reply:
@spadgett @jhadvig Any ideas on how to do this detection?

My initial idea was to add a new flag that would indicate the need to allow localhost CSP sources, i.e. during Console local development.

Reply:
right, I think that should be the path forward